app/controllers/ownerships_controller.rb

```diff
@@ -44,6 +44,58 @@ def destroy
     @ownership.destroy
   end
 
+  # POST /ownerships/batch
+  # {
+  #   "ownership": [
+  #      {
+  #         "model_id": "xxxxxxxxxxx",
+  #         "model_type": "THING",
+  #         "owner_id": "jeff@jeff.com"
+  #      },
+  #      ...
+  #   ]
+  # }
+  def create_batch
+    # Batch upsert does not seem to be possible.
+    # Instead, separate out the creates and updates.
+    #   Batch update does not seem to be possible.
+    #   However, batch create is definitely possible.
+    modelidhash = {}
+    # I don't know what the expected behaviour is if the same model id is passed twice,
+    #  so abort if a duplicate is found.
+    params.require(:ownership).each do |item|
+      modelid = item.require(:model_id)
+      if modelidhash.has_key?(modelid)
+        error = {
+          status: 400,
+          message: "Repeated model id: #{modelid.inspect}"
+        }
+        return render json: error, status: :bad_request
```

Comment:
If you separate parameter processing in a different method don't move this logic into it, instead you can keep the condition for this repeated model error inside this action of the controller (as it is now).